shap/explainers/tree.py

```diff
@@ -533,6 +544,20 @@ def __init__(self, model, data=None, data_missing=None):
             if model.objective is None:
                 self.objective = "squared_error"
                 self.tree_output = "raw_value"
+        elif str(type(model)).endswith("lightgbm.sklearn.LGBMRanker'>"):
+            assert_import("lightgbm")
```

Comment:
quite a bit of this code seems to repeat, might be good to refactor some of the common code out, but I think it's outside the scope of this PR

Reply:
I agree. My first priority was to make sure that the different options were largely independent, so changing one didn't break another one. But that has left some redundant code we could clean up

Reply:
I'll leave this outside the scope of this PR